x-pack/plugin/searchable-snapshots/qa/rest/src/test/resources/rest-api-spec/test/stats.yml

```diff
@@ -0,0 +1,158 @@
+---
+setup:
+
+  - do:
+      indices.create:
+        index: index
+        body:
+          settings:
+            number_of_shards:   1
+
+  - do:
+      bulk:
+        body:
+          - index:
+              _index: index
+          - field: foo
+          - index:
+              _index: index
+          - field: bar
+          - index:
+              _index: index
+          - field: baz
+
+  - do:
+      snapshot.create_repository:
+        repository: repository-fs
+        body:
+          type: fs
+          settings:
+            location: "repository-fs"
+
+  - do:
+      snapshot.create:
+        repository: repository-fs
+        snapshot: snapshot
+        wait_for_completion: true
+
+  - do:
+      snapshot.delete_repository:
+        repository: repository-fs
+
+  - do:
+      indices.delete:
+        index: index
+
+---
+"Tests searchable snapshots stats":
+  - skip:
+      version: " - 7.99.99"
+      reason:  searchable snapshots introduced in 8.0
+
+  - do:
+      searchable_snapshots.stats: {}
+
+  - length: { indices: 0 }
+
+  - do:
+      catch:   missing
+      searchable_snapshots.stats:
+        index: "unkown"
+
+  - do:
+      snapshot.create_repository:
+        repository: repository-searchable-snapshots
+        body:
+          type: searchable
+          settings:
+            delegate_type: fs
+            location: "repository-fs"
+
+  - match: { acknowledged: true }
+
+  - do:
+      snapshot.restore:
+        repository: repository-searchable-snapshots
+        snapshot: snapshot
+        wait_for_completion: true
+
+  - match: { snapshot.snapshot: snapshot }
+  - match: { snapshot.shards.failed: 0 }
+  - match: { snapshot.shards.successful: 1 }
+
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        index: index
+        body:
+          query:
+            match_all: {}
+
+  - match: { hits.total: 3 }
```

Comment:
Does this need us to have refreshed and flushed before taking the snapshot? Or is that handled by the framework?

Reply:
Shards [are flushed right before being snapshotted](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java#L340-L341) so I don't think this is necessary.

Reply:
So they are. Don't mind me...